feature/ingestion

jesse-bos-dij opened the pull request:
### Title
Ingestion logging: traces & generations

### Summary
Adds a simple API for Langfuse `/api/public/ingestion` to log traces and generations.

### Changes
- New `Ingestion` class with:
  - `trace`
  - `generation`
- Batch payload to `/api/public/ingestion`
- Updated README
- Added feature tests

### Backwards compatibility
- No breaking changes, as `Ingestion` is entirely new.
- Broadened prompt `config` typing to `array<string, mixed>`

jesse-bos committed:
Update unit test command in composer.json to remove coverage requirement

jesse-bos committed:
Add Ingestion class for handling trace and generation events

This commit introduces the Ingestion class, which implements methods for tracing and generating events. It utilizes a TransporterInterface for posting data to an API endpoint and includes error handling for unexpected response statuses. The class adheres to strict typing and follows Laravel's best practices.

jesse-bos committed:
Refactor Langfuse class to improve constructor formatting and add ingestion method

This commit updates the README to include details about the new ingestion feature, allowing for the sending of traces and generations via the async `/api/public/ingestion` endpoint. Additionally, the Langfuse class is modified to introduce an `ingestion` method, which initializes the Ingestion class with the transporter and environment parameters, adhering to Laravel's best practices.This commit updates the Langfuse class by enhancing the constructor formatting for the Prompt instantiation and introducing a new ingestion method. The ingestion method allows for specifying the environment while maintaining adherence to strict typing and Laravel's best practices.

jesse-bos committed:
Update README to document new ingestion feature and improve usage examples

This commit enhances the README by adding details about the ingestion feature, which allows sending traces and generations via the async `/api/public/ingestion` endpoint. Additionally, it updates usage examples to utilize named parameters for better clarity and adherence to PHP 8.2+ features.

jesse-bos-dij committed:
Update README to enhance feature documentation and clarify usage

jesse-bos-dij committed:
Refactor Ingestion class to use mixed type hinting for callback functions

This commit updates the Ingestion class to utilize mixed type hinting for the callback functions in the array filtering logic, enhancing type safety and adhering to PHP 8.2+ features.

jesse-bos-dij committed:
Add feature tests for Ingestion class to validate trace and generation functionality

jesse-bos-dij committed:
Rename Langfuse test to prompt test

jesse-bos-dij committed:
Update Prompt and Response classes to use mixed type hinting for config property

This commit modifies the config property type hinting in the Prompt and Response classes from array<int, string> to array<string, mixed>, enhancing type flexibility and adhering to PHP 8.2+ features.

tychoengberinkDIJ commented on README.md:
````diff
@@ -18,16 +24,9 @@ This package provides a wrapper around the [Langfuse](https://langfuse.com) Api,
 
-🤙 Modern codebase , refactoring and static analysis in one command
-```bash  
-composer codestyle  
````
Whats the purpose of deleting this?

jesse-bos-dij replied:
The README is primarily intended for users of the package, so it focuses on installation and usage instructions. Development- and contributor-specific commands like composer codestyle could go under a separate Contributing section or in a dedicated CONTRIBUTING.md. Do you prefer keeping them in the README under a separate heading, or moving them to a separate file?

tychoengberinkDIJ replied:
Lets make a separate file

tychoengberinkDIJ commented on src/Ingestion.php:
```diff
@@ -0,0 +1,113 @@
+    ): void {
+        $envelope = [
+            'id' => $this->uuid(),
+            'timestamp' => gmdate('c'),
```
Cant we just do date('c')

tychoengberinkDIJ commented on src/Ingestion.php:
```diff
@@ -0,0 +1,113 @@
+            'body' => array_filter([
+                'id' => $traceId,
+                'environment' => $this->environment,
+                'timestamp' => gmdate('c'),
```
Cant we just do date('c')

tychoengberinkDIJ commented on src/Ingestion.php:
```diff
@@ -0,0 +1,113 @@
+    ): void {
+        $envelope = [
+            'id' => $this->uuid(),
+            'timestamp' => gmdate('c'),
```
Cant we just do date('c')

tychoengberinkDIJ commented on src/Ingestion.php:
```diff
@@ -0,0 +1,113 @@
+                'id' => $this->uuid(),
+                'traceId' => $traceId,
+                'environment' => $this->environment,
+                'timestamp' => gmdate('c'),
```
Cant we just do date('c')

tychoengberinkDIJ commented on src/Ingestion.php:
```diff
@@ -0,0 +1,113 @@
+        ?array $metadata = null,
+    ): void {
+        $envelope = [
+            'id' => $this->uuid(),
```
I would like to set a custom id if possible

jesse-bos-dij committed:
Enhance CONTRIBUTING.md with detailed guidelines for development setup, code quality, testing, and pull request process

jesse-bos-dij committed:
Refactor Ingestion class to allow optional ID and generation ID parameters

This commit updates the Ingestion class to accept optional `id` and `generationId` parameters in the trace and generation methods. It also replaces `gmdate('c')` with `date('c')` for timestamp generation, enhancing flexibility and adhering to PHP 8.2+ features.
